pkg/plugins/client.go

```diff
@@ -40,13 +40,19 @@ func NewClient(addr string, tlsConfig tlsconfig.Options) (*Client, error) {
 
 	protoAndAddr := strings.Split(addr, "://")
 	sockets.ConfigureTCPTransport(tr, protoAndAddr[0], protoAndAddr[1])
-	return &Client{&http.Client{Transport: tr}, protoAndAddr[1]}, nil
+
+	scheme := protoAndAddr[0]
+	if scheme != "https" {
+		scheme = "http"
+	}
+	return &Client{&http.Client{Transport: tr}, scheme, protoAndAddr[1]}, nil
 }
 
 // Client represents a plugin client.
 type Client struct {
-	http *http.Client // http client to use
-	addr string       // http address of the plugin
+	http   *http.Client // http client to use
+	scheme string       // scheme protocol of the plugin
```

Comment:
yep, there are other approaches. The implications of parsing urls can be different, for instance with semantics between unix socket urls and http urls, so I decided to go with the simplest solution that fixes the problem.